File: src/main/scala/com/vc/hard/L1153.scala
package com.vc.hard

object L1153 {
  def canConvert(str1: String, str2: String): Boolean = {
    import scala.collection.mutable
    val map = new mutable.HashMap[Char, Char]()
    val n = str1.length
    val m = str2.length
    if(str1 == str2) return true
    if(str2.distinct.length >= 26) return false
    if(n != m) return false
    str1.indices.foreach(i => {
      val ch1 = str1(i)
      val ch2 = str2(i)
      if(map.contains(ch1) && map(ch1) != ch2) return false
      map.put(ch1, ch2)
    })
    true
  }
}

File: src/main/java/com/vc/hard/L1153.java
package com.vc.hard;

import java.util.*;

class L1153 {
    public boolean canConvert(String str1, String str2) {
        int n1 = str1.length(), n2 = str2.length();
        if(n1 != n2) return false;

        HashSet<Character> set = new HashSet<>();
        HashMap<Character, Character> map = new HashMap<>();
        for(int i = 0; i < n1; i++) {
            char ch1 = str1.charAt(i), ch2 = str2.charAt(i);
            set.add(ch2);
            if(map.containsKey(ch1) && map.get(ch1) != ch2) return false;
            map.put(ch1, ch2);
        }
        return set.size() < 26 || str1.equals(str2);
    }
}
